js/store.js

```diff
@@ -157,6 +306,37 @@ const store = {
     }
   },
 
+  // Checks if third party is on whitelist for first party,
+  // and if the whitelisted third party should be unwhitelisted.
+  whiteListCheck(origin, target, firstParty, thirdParty) {
+    let newThirdParty = false;
+    if (!('isVisible' in thirdParty) || thirdParty['isVisible']) {
+      // third party needs a whitelist check
+      if (this.onWhiteList(origin, target) && !('isVisible' in thirdParty)) {
+        // third party is whitelisted for this first party
+        // ignore the link for now
+        thirdParty['isVisible'] = false;
+        thirdParty['whiteListedFirstParty'] = origin;
+      } else {
+        // third party is not whitelisted for this first party
```

Comment:
Yes using the updated logic that `isVisible` is present for all `thirdParty` website objects. Previously I had only added the key if a third party was to be ignored.